### ISSUE TITLE
Support of .ico file

### ISSUE THREAD

**Issue by kavitakanojiya** [follmann/favicon_maker#21]
Hey @follmann ,

When I generate them from .png file, this generates files like `android-chrome-72x72.png`, `android-chrome-96x96.png`, etc .

I noticed some different behaviour while generating favicons out of .ico file. It generates files like `android-chrome-96x96-0.png`, ` android-chrome-96x96-1.png`, `android-chrome-96x96-2.png`, etc. Hence, method `#each_icon` lists file named as `android-chrome-96x96.png`. 

Check this:
![screen shot 2015-08-06 at 3 23 50 pm](https://cloud.githubusercontent.com/assets/1167590/9108915/2fd38352-3c4f-11e5-8675-cdce937bbc37.png)

Am I missing anything to generate appropriate favicons out of .ico file ? I have initialized the code the way mentioned in Readme. 

Here is the snippet:

`
    Dir.mktmpdir(Time.now.to_i.to_s) do |dir|
      FaviconMaker.generate do
        setup do
          template_dir  File.dirname(file)
          output_dir    dir
        end

        from File.basename(file) do
          icon 'android-chrome-96x96.png'
          icon 'android-chrome-72x72.png'
          icon 'android-chrome-48x48.png'
          icon 'android-chrome-36x36.png'
          icon 'android-chrome-144x144.png'
          icon 'android-chrome-192x192.png'

          icon 'favicon.png', size: '16x16'

          icon 'apple-touch-icon-57x57.png'
          icon 'apple-touch-icon-60x60.png'
          icon 'apple-touch-icon-72x72.png'
          icon 'apple-touch-icon-76x76.png'
          icon 'apple-touch-icon-114x114.png'
          icon 'apple-touch-icon-120x120.png'
          icon 'apple-touch-icon-144x144.png'
          icon 'apple-touch-icon-152x152.png'
          icon 'apple-touch-icon-180x180.png'
          icon 'apple-touch-icon-precomposed.png', size: '180x180'
          icon 'apple-touch-icon.png', size: '180x180'

          icon 'favicon-16x16.png'
          icon 'favicon-32x32.png'
          icon 'favicon-96x96.png'
          icon 'favicon-194x194.png'

          icon 'favicon.ico', size: '64x64,32x32,24x24,16x16'

          icon 'mstile-144x144.png', format: 'png'
          icon 'mstile-150x150.png', format: 'png'
          icon 'mstile-310x310.png', format: 'png'
          icon 'mstile-70x70.png',   format: 'png'
          icon 'mstile-310x150.png', format: 'png'
        end

        each_icon do |filepath|
          Favicon.create(
            app:        app,
            dimension:  filepath.scan(/\d+x\d+/).first,
            document:   File.open(filepath),
            identifier: 'image' )
        end
      end
    end
`

Any suggestion will be helpful.

Thanks.

**Comment by kavitakanojiya** [comment]
@follmann ,

Image magick breaks down .ico file into multiple resolution supported files. So I tried to make a patch to support .ico to .png file. Here is the compare:

https://github.com/follmann/favicon_maker/compare/master...kavitakanojiya:master.

Can you take a look?

I tried to convert .ico file to .png file. Now I see for circle image, it cuts off the edges. Will you please shed light how to prevent this ?

Thanks

**Comment by follmann** [comment]
Hi Kavita, sorry for my delayed answer...

At first I thought support for ico generation was broken, but as I just checked it isn't. As for using ico-files as a template, this seems kind of backwards to me and I would rather not support it. ico is proprietary and quite limited format. I would rather suggest you use SVG or PNG as template files to achieve good results.

Thanks,

Andreas

**Comment by kavitakanojiya** [comment]
@follmann , thanks for your reply. I went through the documentation for .ico files and it seems to be little complex, however I managed to convert .ico to .png files.

One observation:
InputValidator validates .png and .ico but the specs covers .svg and .png formats. Seems this needs to be fixed if I look into codebase and specs. Makes sense only .svg and .png should be supported, even most of the external services doesnot support .ico file.

Thanks for your inputs.
